### ISSUE TITLE
Use rapidtide to fix potential inflation in FC

### ISSUE THREAD

**Issue by acionca** [TheAxonLab/hcph-sops#207]
See this [paper](https://www.biorxiv.org/content/10.1101/2023.09.08.556939v2) on potential inflation of the whole brain functional connectivity that could be fixed with that [tool](https://github.com/bbfrederick/rapidtide).

**Comment by oesteban** [comment]
One question I have is whether this is also a problem with our *inverse covariance* approach to FC - I have to study the preprint. Have you figured this out @celprov @acionca ?

**Comment by acionca** [comment]
I will have a more detailed look in the following days. I'll keep you posted.

**Comment by acionca** [comment]
#### TL;DR
We should fix the inflation of functional connectivity with [_rapidtide_](https://github.com/bbfrederick/rapidtide/tree/v2.6.3) if we:
- want to look a dynamic functional connectivity
- believe (or observe) that brain activity (in response to stimuli) is higher in the later parts of the tasks runs (this can easily be checked with the toolbox)

Otherwise, I do not believe that it is necessary. However, I plan on denoising a couple of sessions and assesing the difference both in the timeseries and FC matrices anyway. More details below.

#### Here's what I understood from the preprint summarized in a few sentences: 
They reported that functional connectivity (FC) increases over time (within a single run) which is mirrored by an increase in the global mean signal (GMS) vairance. They also reported that this inflation is likely due to an increase in the strength over time of non-neuronal systemic low frequency oscillation (sLFO).

#### What's the issue ?
FC appears to be increasing over the course of an fMRI acquisition. This could lead to inflation of i) computations of dynamic functional connectivity, ii) significant brain activation during a task (especially if included after baseline rest blocs). They report that this inflation is negligible on _between-scan reproducibility and subject discriminability_, especially when looking at time-averaged functional connectivity.


#### How to fix this ?
They provide an easy way of fixing this issue through the [_rapidtide_ toolbox](https://github.com/bbfrederick/rapidtide/tree/v2.6.3). The idea is to _regress out the sLFO with voxel specific delays_ in order to _attenuate the GMS_ variance (see [docs](https://github.com/bbfrederick/rapidtide/tree/v2.6.3)). Practically, the toolbox can be used through docker, yielding the denoised bold volumes along with a bunch of other metrics (quite a lot) to assess the "strength" of sLFO within the data. It is easy and kinda quick to use except that it only takes one single nifti file as input (will require some folder manipulation as it outputs > 70 files for a single input).

#### Do we have to fix it ?
They clearly say that "_FC Inflation Has Negligible Impact on Between-Scan Reproducibility and Subject Discriminability_" and that the fixes from _rapidtide_ produced _only small changes in between-scan reproducibility and ICC-indexed subject discriminability_. From this, I believe it is ok not to fix the resting state if we want to do functional connectome fingerprinting. Still, it is probably safe to check if running _rapidtide_ induces substantial changes in the rs-fc and timeseries. For the task, I believe we should have a look at the brain activity and see if responses to stimuli inflate over the course of the run. The variance explained by sLFO could also be assessed using _rapidtide_. Depending on what we observe, we might add this additional denoising to the pipeline.

**Comment by celprov** [comment]
@acionca Thanks for summarizing the issue.
One element I'm worried about is that this paper also reported that group brain-wide average FC increases across runs (cf Fig. 1b) and showed that group brain-wide average FC is significantly higher in the second session compared to the first one. I'm wondering if that fact coupled with the observation that the inflation rate is not homogeneous across brain regions can bias the images that we acquire towards the end of the session (or more worryingly bias the images that are acquired in the second session of the same day if we hand up acquiring back-to-back)

In my understanding, if we do not correct for inflation:
1. we assume that the subject experiences the same rate of FC inflation in each session (which seems a reasonable assumption given that in the supplementary material that showed between acquisition reproducibility remained high)
2. We strictly stick to comparing images from the same runs (which might not strictly be the case because we stated in the pre-registration that we will compare BOLD runs with and without naturalistic movie)
3. We assume that as long as the subject goes out of the scanner even for a short time, it "resets" the FC inflation.

Given that we do not seem to strictly respect point 2, and we have no evidence regarding point 3, I think it would be worth investigating the correction if our time allows.

**Comment by acionca** [comment]
Thanks Céline - you're absolutely right. I plan on checking anyway how does the correction influence the FC matrices especially when comparing two successive sessions. I'll work on implementing the fix with _rapidtide_ as well as some data management and I'll keep you posted for the results.
